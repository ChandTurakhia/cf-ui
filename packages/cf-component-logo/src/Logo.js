import PropTypes from 'prop-types';
import { createComponent } from 'cf-style-container';
import logoSvg from './logo.svg';
import logoSvgMobile from './logo_mobile.svg';

const Logo = createComponent(({ logoUrl, logoUrlMobile }) => ({
  width: 80,
  height: 40,
  verticalAlign: 'middle',
  background: `transparent url('${logoUrlMobile}') center no-repeat`,
  marginTop: 8,
  marginRight: 5,
  tablet: {
    width: 133,
    background: `transparent url('${logoUrl}') center no-repeat`
  }
}));

Logo.propTypes = {
  logoUrl: PropTypes.string,
  logoUrlMobile: PropTypes.string
};

Logo.defaultProps = {
  logoUrl: logoSvg,
  logoUrlMobile: logoSvgMobile
};

export default Logo;
