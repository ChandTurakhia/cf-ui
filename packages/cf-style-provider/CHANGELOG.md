# Change Log

All notable changes to this project will be documented in this file.
See [standard-version](https://github.com/conventional-changelog/standard-version) for commit guidelines.

<a name="2.2.3"></a>
## [2.2.3](https://github.com/koddsson/cf-ui/compare/cf-style-provider@2.2.2...cf-style-provider@2.2.3) (2017-07-31)




<a name="2.2.2"></a>
## [2.2.2](https://github.com/cloudflare/cf-ui/compare/cf-style-provider@2.2.1...cf-style-provider@2.2.2) (2017-07-19)




<a name="2.2.0"></a>
# [2.2.0](https://github.com/sejoker/cf-ui/compare/cf-style-provider@2.1.2...cf-style-provider@2.2.0) (2017-07-12)


### Features

* **cf-style-provider:** allow passing selectPrefix in cf-style-provider ([3bc1ba9](https://github.com/sejoker/cf-ui/commit/3bc1ba9))




<a name="2.1.4"></a>
## [2.1.4](https://github.com/koddsson/cf-ui/compare/cf-style-provider@2.1.3...cf-style-provider@2.1.4) (2017-07-10)




<a name="2.1.3"></a>
## [2.1.3](https://github.com/koddsson/cf-ui/compare/cf-style-provider@2.1.2...cf-style-provider@2.1.3) (2017-07-10)




<a name="2.1.2"></a>
## [2.1.2](https://github.com/cloudflare/cf-ui/compare/cf-style-provider@2.1.0...cf-style-provider@2.1.2) (2017-07-06)




<a name="2.1.1"></a>
## [2.1.1](https://github.com/koddsson/cf-ui/compare/cf-style-provider@2.1.0...cf-style-provider@2.1.1) (2017-07-06)




<a name="2.1.0"></a>
# [2.1.0](https://github.com/cloudflare/cf-ui/compare/cf-style-provider@2.0.2...cf-style-provider@2.1.0) (2017-07-05)


### Features

* **cf-style-provider:** Print warnings in the dev mode if custom media queries are used ([b8a67c6](https://github.com/cloudflare/cf-ui/commit/b8a67c6))
* **cf-style-provider:** Remove monolithic mode from the dev configuration ([427bf80](https://github.com/cloudflare/cf-ui/commit/427bf80))
* **cf-style-provider:** Sort media queries, add better README ([d89f7fe](https://github.com/cloudflare/cf-ui/commit/d89f7fe))
* **cf-style-provider:** Update snapshots - they use atomic classNames now ([1a42284](https://github.com/cloudflare/cf-ui/commit/1a42284))
