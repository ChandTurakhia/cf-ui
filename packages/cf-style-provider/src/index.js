import createRenderer from './createRenderer';
import felaSnapshot from './felaSnapshot';
import felaTestContext from './felaTestContext';
import StyleProvider from './StyleProvider';

export { createRenderer, felaSnapshot, felaTestContext, StyleProvider };
