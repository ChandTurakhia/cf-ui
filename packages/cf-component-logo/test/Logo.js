import React from 'react';
import renderer from 'react-test-renderer';
import Logo from '../src/Logo';
import { felaSnapshot } from 'cf-style-provider';

test('should render', () => {
  const snapshot = felaSnapshot(<Logo />);
  expect(snapshot.component).toMatchSnapshot();
  expect(snapshot.styles).toMatchSnapshot();
});
