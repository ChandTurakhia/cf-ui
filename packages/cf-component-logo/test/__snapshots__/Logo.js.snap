// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div
  className="a b c d e f g h"
/>
`;

exports[`should render 2`] = `
"
.a {
  width: 80px
}

.b {
  height: 40px
}

.c {
  vertical-align: middle
}

.d {
  background: transparent url('test-file-stub') center no-repeat
}

.e {
  margin-top: 8px
}

.f {
  margin-right: 5px
}

@media (min-width: 47.2em) {
  .g {
    width: 133px
  }
  
  .h {
    background: transparent url('test-file-stub') center no-repeat
  }
}
"
`;
